### PR TITLE
fix(python): Propagate strictness in `from_dicts`

### PR DESCRIPTION
**Pull request by stinodego**
Closes https://github.com/pola-rs/polars/issues/15245

Includes some minor refactoring.

### CODE REVIEW

File: py-polars/src/dataframe/construction.rs [hunk 1]
```diff
@@ -138,48 +135,61 @@ where
...
         let d = d.downcast::<PyDict>()?;
 
-        let mut row = Vec::with_capacity(key_names.len());
-        for k in key_names.iter() {
+        let mut row = Vec::with_capacity(names.len());
+        for k in names.iter() {
             let val = match d.get_item(k)? {
                 None => AnyValue::Null,
-                Some(val) => val.extract::<Wrap<AnyValue>>()?.0,
+                Some(val) => py_object_to_any_value(val, strict)?,
```
Comment (stinodego) [inline py-polars/src/dataframe/construction.rs:149]: This was the key to the issue - calling `extract` on `Wrap<AnyValue>` defaults to strict.

**Comment by ruomad** [comment]
Hello, it doesn't seem to have fully fixed this problem, I still get errors either with PyDataFrame.from_dict or PyDataFrame.from_rows.
It does work as expected when constructing from columns
I've tried with 0.20.22rc1

**Comment by ruomad** [comment]
To be complete it does not throw the same error than with strict = True (in column orientation) but a more cryptic 
<pre>could not append value: "" of type: str to the builder; make sure that all rows have the same schema or consider increasing `infer_schema_length`</pre>

**Comment by stinodego** [comment]
@ruomad Could you give a code example?

**Comment by ruomad** [comment]
<pre>
pl.DataFrame([{'a': 1.0, 'b': ''}], schema = {'a': pl.Float64, 'b': pl.Float64}, orient = 'row', strict = False)
pl.DataFrame([[1.0, '']], schema = {'a': pl.Float64, 'b': pl.Float64}, orient = 'row', strict = False)
</pre>
both yield :
<pre>
polars.exceptions.ComputeError: could not append value: "" of type: str to the builder; make sure that all rows have the same schema or consider increasing `infer_schema_length`
</pre>

**Comment by stinodego** [comment]
@ruomad right, so this has to do with empty strings probably. Could you open an issue and tag me please?

**Comment by ruomad** [comment]
Done

https://github.com/pola-rs/polars/issues/15882

**Pull request merged**